QtAcademyProject.cpp

```diff
@@ -42,7 +175,68 @@ void QtAcademyProject::setMapView(MapQuickView* mapView)
   }
 
   m_mapView = mapView;
+
   m_mapView->setMap(m_map);
 
+  // Start the location display and center the map on the user
+  m_mapView->locationDisplay()->start();
+  m_mapView->locationDisplay()->setAutoPanMode(LocationDisplayAutoPanMode::Recenter);
+
+  // Allow the user to rotate the map by pinching a touchscreen
+  m_mapView->setRotationByPinchingEnabled(true);
+
+  // Create and add a GraphicsOverlay to display Graphics on the MapView
+  GraphicsOverlay* graphicsOverlay = new GraphicsOverlay(this);
+  m_mapView->graphicsOverlays()->append(graphicsOverlay);
+
+  // Create and add a Graphic to show the user's path
+  m_offlineMapExtentGraphic = new Graphic(this);
+  m_offlineMapExtentGraphic->setSymbol(new SimpleLineSymbol(SimpleLineSymbolStyle::Solid, Qt::blue, 2, this));
+  graphicsOverlay->graphics()->append(m_offlineMapExtentGraphic);
+
+  // Create a PolylineBuilder to construct lines given a set of points
+  m_lineBuilder = new PolylineBuilder(SpatialReference::wgs84(), this);
+
+  // Whenever the user's location changes, update the line path if necessary and recenter the map
+  connect(m_mapView->locationDisplay(), &LocationDisplay::locationChanged, this, [this](const Location& l)
+  {
+    if (!m_isTracking)
+      return;
+
+    m_mapView->locationDisplay()->setAutoPanMode(LocationDisplayAutoPanMode::CompassNavigation);
+
+    m_lineBuilder->addPoint(l.position());
+    m_offlineMapExtentGraphic->setGeometry(m_lineBuilder->toGeometry());
+  });
+
   emit mapViewChanged();
 }
+
+int QtAcademyProject::downloadProgress()
+{
+  return m_downloadProgress;
+}
+
+bool QtAcademyProject::isTracking() const
+{
+  return m_isTracking;
+}
```

Comment:
This is optional for both, but you could really use some newer Qt features here to showcase some cool things.

You could use the `MEMBER` `Q_PROPERTY` attribute, which means you don't even need these getters. It's not foolproof and you need to be sure to emit when these change (which you likely are already doing).

Or, you could use the newer C++ property binding stuff. I've never used it and it's new with Qt 6.
https://doc.qt.io/qt-6/bindableproperties.html
https://www.qt.io/blog/property-bindings-in-qt-6